policy/templates/subtemplates/auto/auto-test.gotmpl

```diff
@@ -152,71 +107,70 @@
-          {{- end }}
           confs_dir=./pro-ha
-          env_file=local-{{`${{ matrix.envfiles.db }}`}}.env' >> versions.env
+          env_file=local.env' >> versions.env
+          cat ./confs/{{`${{ matrix.envfiles.config }}`}}.env local-{{`${{ matrix.envfiles.db }}`}}.env > local.env
           echo "::group::versions"
-          cat versions.env
+          cat versions.env local.env
           echo "::endgroup::"
-          # Add Tyk component config variations to $env_file
-          cat confs/{{`${{ matrix.envfiles.config }}`}}.env >> local-{{`${{ matrix.envfiles.db }}`}}.env
           # bring up env, the project name is important
-          docker compose -p auto -f pro-ha.yml -f deps_pro-ha.yml -f {{`${{ matrix.envfiles.db }}`}}.yml -f {{`${{ matrix.envfiles.cache }}`}}.yml {{ if or (eq $test "ui-gate") (eq $test "api-gate") }}-f {{`${{ github.workspace }}`}}/tyk-analytics/tests/ui/ci/Dashboard/compose.yml{{end}} --env-file versions.env --profile master-datacenter up --quiet-pull -d
+          docker compose -p auto -f pro-ha.yml -f deps_pro-ha.yml -f {{`${{ matrix.envfiles.db }}`}}.yml -f {{`${{ matrix.envfiles.cache }}`}}.yml --env-file versions.env --profile master-datacenter up --quiet-pull -d
           ./dash-bootstrap.sh http://localhost:3000
-          docker compose -p auto -f pro-ha.yml -f deps_pro-ha.yml -f {{`${{ matrix.envfiles.db }}`}}.yml -f {{`${{ matrix.envfiles.cache }}`}}.yml {{ if or (eq $test "ui-gate") (eq $test "api-gate") }}-f {{`${{ github.workspace }}`}}/tyk-analytics/tests/ui/ci/Dashboard/compose.yml{{end}} --env-file versions.env --profile slave-datacenter up --quiet-pull -d
+          docker compose -p auto -f pro-ha.yml -f deps_pro-ha.yml -f {{`${{ matrix.envfiles.db }}`}}.yml -f {{`${{ matrix.envfiles.cache }}`}}.yml --env-file versions.env --profile slave-datacenter up --quiet-pull -d
 
-      {{ if or (eq $test "api") (eq $test "api-gate")}}
-      - name: Run tests
-        working-directory: auto
+      {{- if eq .test "api" }}
+      {{- if eq .dot.Name "tyk-analytics" }}
+      - uses: actions/setup-python@v5
+      - working-directory: tests/api
+        run: |
+          pip install -r requirements.txt
+      {{- end }} {{/* tyk-analytics */}}
+      - name: Run API tests
         id: test_execution
         env:
           # Cover pull_request_target too
-        {{- if or (eq $test "ui-gate") (eq $test "api-gate") }}
-          BASE_REF: {{`${{startsWith(github.event_name, 'pull_request') && github.sha || github.ref_name }}`}}
-        {{- else }}
+          {{- if eq .dot.Name "tyk-pro" }}
+          BASE_REF: master
+          {{- else }}
           BASE_REF: {{`${{startsWith(github.event_name, 'pull_request') && github.base_ref || github.ref_name }}`}}
-        {{- end }}
+          {{- end }}
         run: |
           # Generate report id
           echo "id=$(date +%s%N)" >> $GITHUB_OUTPUT
+          mkdir /app && ln -s test/api/pytest_ci.ini /app/pytest_ci.ini
```

Comment:
@alephnull is not possible to write over /app on github runner due to permission error. We may need to decouple or add if statement to diferentiate between local and inside container pytest command

Reply:
I changed your implementation to make it self-contained within the templates.